composer/trainer/trainer.py

```diff
@@ -2142,6 +2158,10 @@ def _train_batch(self, use_grad_scaling: bool) -> Dict[str, torch.Tensor]:
                                     xm.optimizer_step(optimizer, barrier=True)
                                 else:
                                     optimizer.step()
+                # Raise error if automicrobatching and num_alloc_retries increased, as thrashing
```

Comment:
Is error correct here? I'm just wondering how sure we are that _any_ retries is _always_ an error.

Reply:
if we're not sure, we could add a trainer arg `num_cuda_alloc_retries_allowed` or whatever, default it to zero. that way at least a user can escape this error if they want.

Reply:
+1 to this, I think we might want some loose heuristic or something softer than just has an alloc_retry to raise an error. I've seen instances where alloc_retries just happens once during training and it's fine otherwise (so there's borderline no throughput impact).

Reply:
<img width="299" alt="image" src="https://user-images.githubusercontent.com/15602962/221233442-236e0023-ada2-4158-a3bb-bf76dac0f295.png">
e.g. this model hit alloc_retry once, and never hit it again.